docs/elasticsearch/cluster.md

```diff
@@ -23,11 +23,13 @@ The following table provides brief descriptions of the node types.
 
 Node type | Description
 :--- | :--- |
-`Master` | Manages the overall operation of a cluster and keeps track of the cluster state.
-`Master-eligible` | Available for election to become a new master node.
-`Data` | Stores and searches data.
-`Ingest` | Preprocesses data before storing it in a cluster.
-`Coordinating` | Routes search and aggregation requests.
+`Master` | Manages the overall operation of a cluster and keeps track of the cluster state. This includes creating and deleting indices, keeping track of the nodes that join and leave the cluster, checking the health of each node in the cluster (by running ping requests), and allocating shards to nodes.
+`Master-eligible` | Available for election to become a new master node. Elects one node among them as the master node through a voting process.
+`Data` | Stores and searches data. Performs all data-related operations (indexing, searching, aggregating) on the shards stored locally. They are the worker nodes of your cluster and need more disk space than any other node type.
+`Ingest` | Preprocesses data before storing it in a cluster. Runs an ingest pipeline that transforms your data before adding it to an index.
+`Coordinating` | Routes search and aggregation requests. Forwards each client request to the shards on the data nodes, collects and aggregates the results into one final result, and sends this result back to the client.
+
+By default, each node is a master-eligible, data, ingest, and coordinating node. On large clusters, it’s important to have dedicated nodes for each role for the overall stability of the cluster.
```

Comment:
We've got to be way more specific here.

For example: three dedicated master nodes is right for all/just about all production use cases. We should mention that the way to achieve a dedicated node type is to mark everything else as false. In this case, you also have to mark all the other nodes as not master-eligible.

When is a coordinating node appropriate? A search-heavy workload? What's the appropriate ratio of coordinating nodes to data nodes? Should they be more CPU heavy, compared to more storage and RAM heavy for data nodes? If I don't do any transformation at all, do I need dedicated ingest nodes? (Seems unlikely). If I deploy across three zones, should my data nodes also be in a multiple of three? (Probably.) What about my coordinating or ingest nodes? Is it fine to have just 1-2 of them even if I'm spread across three zones? (Probably...?)